src/rfc2047.rs

```diff
@@ -157,4 +172,16 @@ mod tests {
             assert!(decode_rfc2047(*t).is_none());
         }
     }
+
+    #[test]
+    fn test_encode_preserves_ascii_string() {
+        let plain_ascii = "uiae 282!";
+        assert_eq!(encode_rfc2047(plain_ascii), plain_ascii);
+    }
+
+    #[test]
+    fn test_encode_uses_base64_for_non_ascii_string() {
+        let umlauty = "Überräschùng!";
+        assert_eq!(&encode_rfc2047(umlauty)[..10], "=?utf-8?B?");
+    }
```

Comment:
There should be a test, I think, which decodes the encoded string to make sure it is well-formed (or at least well-formed enough for the decoder in this crate).